Dockerfile.development

```diff
@@ -4,46 +4,27 @@ FROM dependabot/dependabot-core
 USER root
 
 # Add debugging tools
-RUN apt-get update && apt-get install -y vim strace ltrace gdb
+RUN apt-get install -y vim strace ltrace gdb
 
-# This Dockerfile adds a non-root 'dependabot' user. However, for Linux,
-# this user's GID/UID must match your local user UID/GID to avoid permission issues
-# with bind mounts. Update USER_UID / USER_GID if yours is not 1000.
-# USER_UID should be the value of $UID from the host environment and
-# USER_GID the value of `id -g`.
-# See https://aka.ms/vscode-remote/containers/non-root-user for details.
-ARG USER_UID=1000
-ARG USER_GID=$USER_UID
-ARG USERNAME=dependabot
+USER dependabot
+RUN git config --global user.name dependabot-ci \
+  && git config --global user.email no-reply@github.com
 
-RUN groupadd -o --gid "${USER_GID}" "${USERNAME}" && \
-  useradd --uid "${USER_UID}" --gid "${USER_GID}" -m "${USERNAME}"
-RUN chown -R "${USERNAME}":"${USERNAME}" \
-  /usr/local/.pyenv \
-  /opt/go/gopath \
-  /opt/rust/
-USER $USERNAME
-RUN git config --global user.name dependabot-ci && git config --global user.email no-reply@github.com
-
-ARG CODE_DIR=/home/$USERNAME/dependabot-core
+ARG CODE_DIR=/home/dependabot/dependabot-core
 
 RUN curl -L -o ~/.vimrc https://github.com/hmarr/dotfiles/raw/main/vimrc-vanilla.vim && \
   echo 'export PS1="[dependabot-core-dev] \w \[$(tput setaf 4)\]$ \[$(tput sgr 0)\]"' >> ~/.bashrc
 
-RUN mkdir -p ${CODE_DIR}/common/lib/dependabot
-WORKDIR ${CODE_DIR}
-
-ENV BUNDLE_PATH="/home/$USERNAME/.bundle" \
+ENV BUNDLE_PATH="/home/dependabot/.bundle" \
   BUNDLE_BIN=".bundle/binstubs"
 ENV PATH="$BUNDLE_BIN:$PATH:$BUNDLE_PATH/bin"
 
-COPY common/Gemfile common/dependabot-common.gemspec ${CODE_DIR}/common/
-COPY common/lib/dependabot/version.rb ${CODE_DIR}/common/lib/dependabot/
-RUN cd common && bundle install
+WORKDIR ${CODE_DIR}
```

Comment:
```suggestion
```

Concern: this is a problem if `DOCKER_BUILDKIT=0`, and I'm not sure we should require that...

I had to check, but before BuildKit the `WORKDIR` command ignores the current `USER`:
```
$ cat Dockerfile
FROM debian:buster
WORKDIR /tmp/a-workdir-pre
USER nobody
WORKDIR /tmp/a-workdir-post
RUN touch /tmp/a-file

$ DOCKER_BUILDKIT=0 docker build -t foo . && docker run --rm foo ls -l /tmp
total 8
-rw-r--r-- 1 nobody nogroup    0 Apr  7 17:01 a-file
drwxr-xr-x 2 root   root    4096 Apr  7 17:01 a-workdir-post
drwxr-xr-x 2 root   root    4096 Apr  7 17:01 a-workdir-pre

$ DOCKER_BUILDKIT=1 docker build -t foo . && docker run --rm foo ls -l /tmp
total 8
-rw-r--r-- 1 nobody nogroup    0 Apr  7 17:01 a-file
drwxr-xr-x 2 nobody nogroup 4096 Apr  7 17:01 a-workdir-post
drwxr-xr-x 2 root   root    4096 Apr  7 17:01 a-workdir-pre
```

This manifests as L26 failing: since `$CODE_DIR` is owned by `root:root` the `mkdir` lacks permission to write it.

We could:
* Move it after the `mkdir` block
* Prefer absolute paths (e.g. `cd "${CODE_DIR}/common && bundle install`)